add kubernetes ingress controller (close #25) (#102)
megaease/easegress: doc/ingresscontroller.md

      serviceAccountName: easegress-ingress-controller
      containers:
      - command:
        - /bin/sh
        - -c
        - |-
          echo name: $POD_NAME > /easegress-ingress/config.yaml && echo '
          cluster-request-timeout: 10s
          cluster-role: writer
          api-addr: 0.0.0.0:2381
          cluster-name: easegress-ingress-controller
          ' >> /easegress-ingress/config.yaml && /opt/easegress/bin/easegress-server -f /easegress-ingress/config.yaml
        image: megaease/easegress:latest
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        name: easegress-ingress
        volumeMounts:
        - mountPath: /easegress-ingress
          name: ingress-params-volume
      volumes:
      - emptyDir: {}
        name: ingress-params-volume

---
kind: Service
apiVersion: v1
metadata:
  name: easegress-ingress-service
  namespace: default
spec:
  selector:
    app: easegress-ingress
  ports:
    - name: admin
      protocol: TCP
      port: 2381
      nodePort: 2381
    - name: web
      protocol: TCP
      port: 8080
      nodePort: 80
  type: NodePort
```

> The service exposes two ports, the `admin` port is for Easegress administration, we exposed it for simplicity here, but you may not want to expose it in your environment for security; the `web` port is to receive external HTTP requests and forward them to the HTTP server in Easegress.

Reviewer: ```suggestion
The service exposes two ports, the `admin` port is for Easegress administration, we exposed it for simplicity here, for security consideration, we don't recommend you to expose it in your environment; the `web` port is to receive external HTTP requests and forward them to the HTTP server in Easegress.
```
